run_brer/run_config.py

```diff
@@ -583,6 +583,8 @@ def run(self, tpr_file=None, **kwargs):
             else:
                 assert end_time > start_time
                 if end_time - start_time >= requested_production_time:
+                    prev_iter_state_json = 'state_' + str(self.run_data.get('iteration')) + '.json'
+                    shutil.copy2(self.state_json, prev_iter_state_json)
                     self.run_data.set(phase='training',
```

Comment:
Side note: the copy occurs at a point where the state data may be updated in memory, but has not yet been written. It serves as an archive of the previous input, not as a point-in-time snapshot. It might make logical sense to make the archive before starting the iteration instead of after finishing it. However, that data is already available in the regular state file, and making the archive at the end of the iteration serves an extra purpose: it leaves a breadcrumb trail in the filesystem that indicates that the script made it successfully to line 586, if nothing else.
I'm not sure if it is worth documenting this. I'm just "thinking out loud."

Reply:
Are you proposing that we copy the `state_json` to `prev_iter_state_json` before the iteration concludes? Wouldn't this mess with any syntax/functions during the iterations? Or are you suggesting that we copy to `prev_iter_state_json` before the next iteration occurs?

Reply:
Oh gotcha, no it's a valid comment! My thought was that we needed to copy the `prev_iter`'s `state_json` once it's "completed" but not before it's overwritten, hence my placement here before `run_data` resets to training.

Reply:
I agree.

I guess we probably should document (somewhere) the existence *and purpose* of this new file. If there is already a place where the `state.json` file is described, we could mention it there. Otherwise, I think I would like a one-to-two-sentence version of this conversation as a comment line.

Do you mind taking a browse through https://kassonlab.github.io/run_brer/ to see if there's a good place to mention it?

Reply:
Will do!

Reply:
I don't see a spot where `state.json` is explicitly defined. There are a couple spots where `run_brer` or `run_config` uses `state.json` in function definitions. The best spot would be [https://kassonlab.github.io/run_brer/user_api.html#module-run_brer.run_data](https://kassonlab.github.io/run_brer/user_api.html#module-run_brer.run_data) if we were to add the definition to the documentation. Otherwise, we can comment above our proposed changes

Reply:
It looks like it pops up a few times in https://kassonlab.github.io/run_brer/developer_api.html

You could add a sentence to the `run_brer.directory_helper.DirectoryHelper` doc-string and also add a comment line here.

Alternatively, you could move these few new lines to a new function like [`write_to_json`](https://kassonlab.github.io/run_brer/developer_api.html#run_brer.metadata.MultiMetaData.write_to_json) and just mention the behavior there. That might be unnecessary extra work. Whatever you prefer is fine.